bin/FileSystem.js

```diff
@@ -0,0 +1,24 @@
+module.exports = class FileSystem {
+
+  constructor(driver) {
+    this.driver = driver;
+  }
+
+  async writeFile(data, projectRelativeDirectory, filename, reporter = console.log) {
+    const directory = await this.driver.writeDirectory(projectRelativeDirectory);
+    //  Not using path.join as want to keep this layer separate from the file system driver
+    const fullPath = directory + filename;
+
+    return this.driver.writeFileAsync(fullPath, data)
+      .then(() => {
+        reporter(`Written config to ${fullPath}`);
+      })
+      .catch(err => { throw err });
```

Comment:
unrelated, but why are explicit `catch` needed if the only behavior is to re-throw the exception? I thought that was the default but there are too many promises implementations to keep track

Reply:
Unhandled Promise rejections are deprecated in node. Although it won't make a material difference in this particular case, as you suggest, trying to be a good citizen.